### DISCUSSION TITLE
Using func.any_value() causes custom column type to be bypassed

### DISCUSSION THREAD

**Discussion by huntfx** [sqlalchemy/sqlalchemy#6438] [Usage Questions] [answered]
**Describe the bug**
Using [`ANY_VALUE`](https://stackoverflow.com/a/60005384/2403000) in a `GROUP BY` query will result in any custom column types being bypassed, and instead they revert to their default `impl` value.

```
# Model.date is of ArrowType, which has impl set to DateTime

>>> session.query(func.max(Model.date)).first()
(<Arrow [2021-05-06T11:27:07.792000+01:00]>,)  # Correct

>>> session.query(func.any_value(Model.date)).first()
(datetime.datetime(2021, 5, 6, 10, 27, 7, 792000),)  # Bypassed process_result_value and reverted to DateTime
```

**Expected behavior**
The custom type will be used, and the result will be processed by `process_result_value`.
In the above example, both outputs should be the same.

**To Reproduce**
It'll need a MySQL connection string

``` python
import json
from sqlalchemy import Column, types, create_engine, func
from sqlalchemy.ext.declarative import declarative_base
from sqlalchemy.orm import sessionmaker

class JsonType(types.TypeDecorator):
    impl = types.String
    def process_bind_param(self, value, dialect):
        return json.dumps(value)

    def process_result_value(self, value, dialect):
        if value is None:
            return None
        return json.loads(value)

Engine = create_engine('mysql:///')
Base = declarative_base(bind=Engine)
Session = sessionmaker(bind=Base.metadata.bind)

class Model(Base):
    __tablename__ = 'model'
    id = Column(types.Integer, primary_key=True)
    value = Column(JsonType(1024), nullable=False)

Base.metadata.create_all()

session = Session()
model = Model(value=['a', 'b'])
session.add(model)
session.commit()

print(session.query(func.max(Model.value)).one()[0])
# (['a', 'b'],)
print(session.query(func.any_value(Model.value)).one()[0])
# ('["a", "b"]',)
```

**Versions.**
 - OS: Windows 10
 - Python: 3.7
 - SQLAlchemy: Tested on 1.3 and 1.4
 - Database: MySQL 8.0
 - DBAPI: MySQLdb

**Additional context**
If there's a way to force the type like `func.any_value(Model.value).as_type(ArrowType)`, then that would also be fine to use as a workaround.

**Comment by zzzeek** [comment, marked as answer]
hey there -

lesser-used and special functions [dont usually have return types](https://docs.sqlalchemy.org/en/14/tutorial/data_select.html#built-in-functions-have-pre-configured-return-types) so we have to specify these:

    print(session.query(func.any_value(Model.value, type_=Model.value.type)).one()[0])

**Reply by huntfx** [reply]
Ah ok thanks, good to know the background behind it :)
